flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/endpoint/hive/HiveServer2Endpoint.java

```diff
@@ -310,7 +311,11 @@ public TOpenSessionResp OpenSession(TOpenSessionReq tOpenSessionReq) throws TExc
             // all the alive PersistenceManager in the ObjectStore, which may get error like
             // "Persistence Manager has been closed" in the later connection.
             hiveCatalog.open();
-            Module hiveModule = new HiveModule();
+            // create hive module lazily
+            FunctionCreator<Module> hiveModuleCreator =
+                    (options, readableConfig, classLoader) ->
+                            FactoryUtil.createModule(
```

Comment:
Notice: Factory#createModule still uses the AppClassloader to find the Factory. Please take at look at the `FactoryUtil#createModule` Line 486

Reply:
It uses the classloader in Context, and so is user classloader.